### DISCUSSION TITLE
Is there any way we can use s3 storage with this ?

### DISCUSSION THREAD

**Discussion by akhzarjaved** [spatie/browsershot#563] [General]
I am willing to use s3 storage with this.
Is there any way we can use s3 storage ?

**Comment by hyphen81** [comment]
Did you ever figure out a way to do this? I need to save a PDF to S3 instead of locally, and I can't seem to get it working with BrowserShot.

**Reply by akhzarjaved** [reply]
I don't remember what i did back then, But a workaround can be:
- Save the file locally then upload to s3 using storage facade

**Reply by bnski** [reply]
In Laravel this seems to work just fine...

```php
Storage::disk('s3')->put('filename.pdf', Browsershot::url('https://')->pdf());
```

**Reply by toriqsetiawan** [reply]
you can use like this for an image

```
      Storage::disk('s3')->put(
          'thumbnail/filename.jpg',
          Browsershot::url($url)
              ->windowSize(1280, 2000)
              ->waitUntilNetworkIdle()
              ->screenshot()
      );
```
